sdlib/d/gc/extent.d

```diff
@@ -306,57 +306,127 @@ public:
 	}
 
 	/**
-	 * Freespace Flag features for slabs.
+	 * Metadata features for slabs.
 	 */
 
 	@property
-	bool supportsFreeSpace() const {
+	bool supportsSlabMetaData() const {
```

Comment:
Is this API used outside of contracts here? I have a hard time imagining it would in a way that's not a bug as the pd also has the size class.

If that's the case, it's probably preferable to inline isAppendableSizeClass in the contract themselves.

It would also be nice to rename this, because it's not just appenable anymore.